src/test/java/software/amazon/cloudwatchlogs/emf/model/MetricsContextThreadSafetyTest.java

```diff
@@ -0,0 +1,52 @@
+package software.amazon.cloudwatchlogs.emf.model;
+
+import static org.junit.Assert.assertEquals;
+
+import java.util.Map;
+import org.junit.After;
+import org.junit.Test;
+
+public class MetricsContextThreadSafetyTest {
+    private volatile Throwable throwable = null;
+
+    @Test
+    public void testConcurrentPutMetaData() throws InterruptedException {
+        MetricsContext mc = new MetricsContext();
+        Thread[] threads = new Thread[100];
+        long targetTimestampToRun = System.currentTimeMillis() + 500;
+
+        for (int i = 0; i < 100; i++) {
```

Comment:
Can we move `100` an `1000` to top of this method as finals (i.e. `N_THREADS` & `N_PUT_METADATA`).

Then below you can `use N_THREADS*N_PUT_METADATA`